text/0000-release-process.md

```diff
@@ -0,0 +1,296 @@
+
+                        +---------------+      +------------------+      +---------+
+                        |  development  | ---> | feature complete | ---> | release |
+                        +---------------+      +------------------+      +---------+
+```
+
+### Terminology
+
+#### User Acceptance Testing
+
+User acceptance testing (UAT) is the process of manually verifying that new features and fixes are **complete** and 
+**don't introduce issues**. Typically, there are two strategies:
+
+1. Rolling UAT - User acceptance is done per issue as the issue is completed.
+2. Feature-Complete UAT - User acceptance is done when all features are complete.
+
+More commonly Feature-Complete User Acceptance is used in practice because any additional changes could have unexpected
+side-effects invalidating prior completed UATs. In our case, Feature-Complete UAT fit perfectly during the
+[feature complete](#feature-complete) phase.
+
+#### Feature Complete
+
+Feature complete for a particular release is a period of time when changes are analyzed and vetted before being added to
+a release.
+
+The following take place:
+
+1. At the beginning, a release branch is created (`release/<version>`).
+2. At the beginning, a [change control board](#change-control-board) is organized on a daily basis to review additional
+requested changes.
+3. During, only changes approved by the CCB are merged into the release branch.
+
+#### Change Control Board
+
+A change control board would consist of core developers that, on a daily basis, organize changes that might
+be suitable for [release branches](#release-branches) during the feature complete.
+
+The board members typically decide on whether a change should be included on the release based on the following criteria:
+
+- Impact
+- Effort
+- Risk
+
+#### Release branches
+
+Release branches are protected branches that will be created at the beginning of feature complete. These branches are
+protected, similar to `master`, if not more so depending on the [change control board](#change-control-board) members. 
+
+The release branch will be merged into `master` upon release.
+
+```text
+                 master
+
+                   o
+                   |
+                   |    release/1.2.3
+                   |
+                   |          o commit: fix/critical-or-minor
+                   |          |
+                   |          |
+commit: feature/Y  o          |
+                   |          |
+                   |          |
+                   |          |
+                   |          |
+                   +----------+ <--+ start feature complete
+                   |
+                   |
+                   |
+                   |
+commit: feature/X  o
+                   |
+                   |
+                   +
+```
+
+#### Migration Guide
+
+A migration guide is a document which details breaking changes for migrating from prior versions.
+
+### Release Execution
```

Comment:
Announcements via e-mail, slack, social media(?) should be included as part of the release plan so we don't forget.

Reply:
Added